File: solutions/2025-dec/30/sum_the_string.js
function stringSum(str) {
    let total = 0;
    let stringStr = "";
    for (const c of str) {
        if (Number.isFinite(Number(c))) {
            stringStr += c;
        } else if (stringStr) {
            total += parseInt(stringStr);
            stringStr = "";
        }
    }
    if (stringStr) total += parseInt(stringStr);
    return total;
}

File: solutions/2025-dec/30/sum_the_string.py
def string_sum(s):
    total = 0
    num_str = ""
    for c in s:
        if c.isdigit():
            num_str += c
        elif not c.isdigit() and num_str:
            total += int(num_str)
            num_str = ""
    if num_str:
        total += int(num_str)
    return total
